pkg/storage/endpoint/key_path.go

```diff
@@ -238,20 +237,10 @@ func KeyspaceGroupIDPath(id uint32) string {
 	return path.Join(tsoKeyspaceGroupPrefix, keyspaceGroupMembershipKey, encodeKeyspaceGroupID(id))
 }
 
-// ExtractKeyspaceGroupIDFromPath extracts keyspace group id from the given path, which contains
-// the pattern of `tso/keyspace_groups/membership/(\d{5})$`.
-func ExtractKeyspaceGroupIDFromPath(path string) (uint32, error) {
+// GetCompiledKeyspaceGroupIDRegexp returns the compiled regular expression for matching keyspace group id.
```

Comment:
Why do we need to separate it into two functions?

Reply:
performance concern. Compile the patten once and use it all the time to boost performance.